### PR TITLE
Ubuntu setup docs

### PR DESCRIPTION
**Pull request by Ry-DS**
Closes https://github.com/UNSWComputing/rUNSWift/issues/2772

I am yet to make the change_field script these docs speak of, but it will behave as its desribed here

### CODE REVIEW

File: source/setup/nao_setup_ubuntu.rst
Comment (Ry-DS) [inline source/setup/nao_setup_ubuntu.rst]: Typo

File: source/setup/nao_setup_ubuntu.rst
Comment (Ry-DS) [inline source/setup/nao_setup_ubuntu.rst]: Rephrase or remove

File: source/setup/nao_setup_ubuntu.rst
Comment (Ry-DS) [inline source/setup/nao_setup_ubuntu.rst]: Don't show downloading old one, no point

File: source/setup/nao_setup_ubuntu.rst
Comment (Ry-DS) [inline source/setup/nao_setup_ubuntu.rst]: A bit confusing with the below text also

File: source/setup/nao_setup_ubuntu.rst
Comment (Ry-DS) [inline source/setup/nao_setup_ubuntu.rst]: And running the cat command

File: source/setup/nao_setup_ubuntu.rst
Comment (Ry-DS) [inline source/setup/nao_setup_ubuntu.rst]: "and then attempt to flash the custom image once you confirm things roughly work as expected in the base image"

File: source/setup/nao_setup_ubuntu.rst
Comment (Ry-DS) [inline source/setup/nao_setup_ubuntu.rst]: Outdated

File: source/setup/nao_setup_ubuntu.rst
Comment (Ry-DS) [inline source/setup/nao_setup_ubuntu.rst]: Not strictly true. Can be done via wifi as well

**Comment by jayenashar** [comment]
is the change field script like setup-game.sh?

**Comment by Ry-DS** [comment]
> is the change field script like setup-game.sh?

It will only set the wifi to the correct field wifi, nothing else. We hope to have all setup steps possible through web nao in the future. Th change field script will act as the de-facto way to change wifi until we have it in web nao (or possibly called by web nao).

setup-game.sh is now a legacy script (deprecated), we won't be making anything like it

File: .gitignore
Comment (pzrq) [inline .gitignore:3]: Nitpick: I prefer a [global gitignore](https://stackoverflow.com/questions/7335420/global-git-ignore) for this as it only needs to be done once per machine for all repos, rather than for every repo.
```suggestion
```

File: source/setup/index.rst
Comment (pzrq) [inline source/setup/index.rst]: Doesn't have to be this PR, though if not, it should be tracked as a new GitHub issue.

My first thought was could `nao_setup` be renamed to `nao_setup_softbank` ? 

Then I thought that's not a good idea, because we aren't keeping it for 2024 onwards at all. So only a brief note is needed, e.g. linking to:
https://runswift.readthedocs.io/en/2023/setup/nao_setup.html

The key is the 2023 branch/version. Then you don't even need a `nao_setup_ubuntu` page, just edit the existing `nao_setup` page.

File: source/setup/nao_setup_ubuntu.rst
Comment (pzrq) [inline source/setup/nao_setup_ubuntu.rst]:
```suggestion
This offers several advantages. On top of being more familiar to most of us, it makes things a lot simpler from a technical perspective,
```

File: source/setup/nao_setup_ubuntu.rst
Comment (pzrq) [inline source/setup/nao_setup_ubuntu.rst]:
```suggestion
alongside allowing us to pursue new technical projects with a better chance of success such as the ongoing research into a migration to ROS/ROS2.
```

File: source/setup/nao_setup_ubuntu.rst
Comment (pzrq) [inline source/setup/nao_setup_ubuntu.rst]: There's a LOT of these. See if there's an autoformatter, a lint rule or something that just does them all for you, I haven't flagged them all as I think it's a style nitpick.


```suggestion
**************
```

File: source/setup/nao_setup_ubuntu.rst
Comment (pzrq) [inline source/setup/nao_setup_ubuntu.rst]:
```suggestion
**************
```

File: source/setup/nao_setup_ubuntu.rst
Comment (pzrq) [inline source/setup/nao_setup_ubuntu.rst]:
```suggestion
Before doing anything, you should build `runswift`. While technically optional, this will be useful later on, as the image
```

File: source/setup/nao_setup_ubuntu.rst
Comment (pzrq) [inline source/setup/nao_setup_ubuntu.rst]:
```suggestion
you create would automatically bake in the ``runswift`` executable if it exists. Make sure you follow the PC setup guide beforehand.
```

File: source/setup/nao_setup_ubuntu.rst
Comment (pzrq) [inline source/setup/nao_setup_ubuntu.rst]:
```suggestion
Use ``docker.ps1`` if you're on Windows.
```

File: source/setup/nao_setup_ubuntu.rst
Comment (pzrq) [inline source/setup/nao_setup_ubuntu.rst]: Just a suggestion, I don't know what the TC does, but I often see things like this to slow down spam bots.
```suggestion
- For teams new to RoboCup SPL, please contact the RoboCup SPL Technical Committee (rc-spl-tc AT lists.robocup.org) as mentioned `here <https://spl.robocup.org/v6-support/>`_.
```

File: source/setup/nao_setup_ubuntu.rst
Comment (pzrq) [inline source/setup/nao_setup_ubuntu.rst]:
```suggestion
- If you're not aligned with RoboCup, please contact your regional NAO reseller, most likely either Softbank Robotics or United Robotics Group.
```

File: source/setup/nao_setup_ubuntu.rst
Comment (pzrq) [inline source/setup/nao_setup_ubuntu.rst]:
```suggestion
If you are on site at UNSW, running ``build-naoimage.sh`` will download the opn file for you as part of the image creation process.
```

File: source/setup/nao_setup_ubuntu.rst
Comment (pzrq) [inline source/setup/nao_setup_ubuntu.rst]: **Blocker**: I don't remember as part of Softbank legal if we can even reveal the existence of this `ROBOCUP_ONLY` file / filename. But it's now public (uh oh - GitHub support request?)

@jayenashar Do you happen to recall so we can unblock this PR at least?

File: source/setup/nao_setup_ubuntu.rst
Comment (pzrq) [inline source/setup/nao_setup_ubuntu.rst]:
```suggestion
****************************
```

File: source/setup/nao_setup_ubuntu.rst
Comment (pzrq) [inline source/setup/nao_setup_ubuntu.rst]:
```suggestion
****************************
```

File: source/setup/nao_setup_ubuntu.rst
Comment (pzrq) [inline source/setup/nao_setup_ubuntu.rst]: Or `docker` if referring to the CLI instead of the app?
```suggestion
Ensure you have Docker installed before continuing. This process will take a significant period if you are running for the first time.
```

File: source/setup/nao_setup_ubuntu.rst
Comment (pzrq) [inline source/setup/nao_setup_ubuntu.rst]:
```suggestion
******************
```

File: source/setup/nao_setup_ubuntu.rst
Comment (pzrq) [inline source/setup/nao_setup_ubuntu.rst]:
```suggestion
******************
```

File: source/setup/nao_setup_ubuntu.rst
Comment (pzrq) [inline source/setup/nao_setup_ubuntu.rst]:
```suggestion
************************************
```

File: source/setup/nao_setup_ubuntu.rst
Comment (pzrq) [inline source/setup/nao_setup_ubuntu.rst]:
```suggestion
- macOS/Linux
```
OR
```suggestion
- Mac/Linux
```

There is no MacOS ;)

File: source/setup/nao_setup_ubuntu.rst
Comment (pzrq) [inline source/setup/nao_setup_ubuntu.rst]:
```suggestion
    * You can find its head id via ``cat /sys/qi/head_id`` after ``ssh <robot>``.
```

File: source/setup/nao_setup_ubuntu.rst
Comment (pzrq) [inline source/setup/nao_setup_ubuntu.rst]:
```suggestion
    * Note you can flash safely without adding the robot to this file, and discover the head ID by ``ssh <IP>`` the robot calls out after flashing, and running the ``cat`` command.
```

File: source/setup/nao_setup_ubuntu.rst
Comment (pzrq) [inline source/setup/nao_setup_ubuntu.rst]: Double check this one, it might have already been there, or it might not have been an issue, I don't recall the exact RST syntax.

```suggestion
at the competition, for example ``SPL_A`` to ``SPL_E`` has been typical of RoboCup.
```

**Review by pzrq — COMMENTED** [review]
Great work!

I think there's just one potential blocker that can hopefully get addressed sooner rather than later.

File: .gitignore
Comment (Ry-DS) [reply .gitignore:3]: yeah but tbh too lazy to put this no all machines I use. .idea is pretty standard so don't see it being an issue to include alongside .vscode, and saves future individuals who are also as lazy as me

File: source/setup/index.rst
Comment (Ry-DS) [reply source/setup/index.rst]: sry don't follow, what do you reckon we change this to?

File: source/setup/nao_setup_ubuntu.rst
Comment (Ry-DS) [reply source/setup/nao_setup_ubuntu.rst]: Pretty sure this was copied from the previous setup guide. Open to whatever is right

File: source/setup/index.rst
Comment (Ry-DS) [reply source/setup/index.rst]: after discussing in person, I understand now. We're going to remove the old one entirely and simply add a link to the old docs (with the link pete provided)

**Review by pzrq — APPROVED** [review]
LGTM 👍🏼 

@Ry-DS I'll let you decide whether you'd like another approval to :shipit: or you're happy hitting the `Merge` (or similar) button yourself.

File: source/setup/nao_setup.rst
Comment (Ry-DS) [inline source/setup/nao_setup.rst]:
```suggestion
You can also provide ``all`` to change_field to change the field of all robots in robots.cfg.
```
pending decision on what the script is called here: https://github.com/UNSWComputing/rUNSWift/pull/2800

Comment (pzrq) [reply source/setup/nao_setup.rst]: I actually like change-field more as I think about it, because then it's not specific to wifi if the SPL changes to ultrasound-based comms, for example.

**Comment by pzrq** [comment]
@Ry-DS I think it'd be good to update to recommend Ubuntu 22.04 LTS while in the area, I don't know if anyone is actually testing on other OSes right now, perhaps you do? 
https://github.com/UNSWComputing/rUNSWift-docs/blob/main/source/contributing/pc_setup_contributing.rst

File: source/setup/nao_setup.rst
Comment (Ry-DS) [reply source/setup/nao_setup.rst]: I'm going to make it change-wifi since that's unlikely to happen for awhile, and change-wifi is more straight forward. We can explore changing it back if this does happen

**Pull request merged**
